### PR TITLE
🏗 Update react-dates build script

### PR DESCRIPTION
**Pull request by alanorozco**
- Scopes to its own package
- Moves dependencies to package:
  - `acorn-globals`
  - `ast-replace`
  - `escodegen`
  - `rocambole`
- Updates created bundle

### CODE REVIEW

File: third_party/react-dates/package.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    "derequire": "^2.1.1",
+    "envify": "^4.1.0",
+    "escodegen": "^2.0.0",
+    "prop-types": "^15.7.2",
+    "react-addons-shallow-compare": "^15.6.3",
+    "react-externs": "^0.13.6",
+    "react-with-direction": "^1.3.1",
+    "rocambole": "^0.7.0",
+    "uglifyify": "^5.0.2"
+  }
```
Comment (rsimha) [inline third_party/react-dates/package.json:26]: Can we use exact versions here to streamline renovate updates? (Removing the carets should suffice.) At that point, I believe these packages will get grouped under `subpackage devDependencies`:

https://github.com/ampproject/amphtml/blob/aa3c1a04d106ae70615eb2ee70e2c1d34a633e5f/.renovaterc.json#L24-L31

**Review by rsimha — APPROVED** [review]
LGTM! Thank you for making things even better 😃

File: package.json [hunk 2]
```diff
@@ -57,6 +57,7 @@
     "@types/minimist": "1.2.1",
     "@types/mocha": "8.2.1",
     "@types/node": "14.14.32",
+    "0x": "4.10.2",
```
Comment (rsimha) [inline package.json:60]: This was inadvertently added by another PR: https://github.com/ampproject/amphtml/pull/33111/files#r591649903. Can we remove it here?

Comment (alanorozco) [reply package.json:60]: Oh, I thought that change was intentional. Removed.

Comment (rsimha) [reply package.json]: This is now being cleaned up in #33306.

**Comment by rsimha** [comment]
`react-dates` was upgraded to v21 in #34073. This PR will need to be rebased and re-tested before it can be merged.

**Pull request closed without merging**